src/lib.rs

```diff
@@ -94,3 +94,51 @@ pub(crate) fn get_field(
 
     Ok(field)
 }
+
+#[macro_export]
+macro_rules! impl_py_copy {
+    ($pyo3_ty:ty) => {
+        #[pymethods]
+        impl $pyo3_ty {
+            pub fn __copy__(&self) -> Self {
+                self.clone()
+            }
+        }
+    };
+}
+
+#[macro_export]
+macro_rules! impl_py_deepcopy {
+    ($pyo3_ty:ty) => {
+        #[pymethods]
+        impl $pyo3_ty {
+            pub fn __deepcopy__(&self, _memo: &::pyo3::types::PyDict) -> Self {
+                self.clone()
+            }
+        }
+    };
+}
+
+#[macro_export]
+macro_rules! impl_py_eq {
```

Comment:
I've added support for equality as well, since it makes it significantly easier to test and compare types.